### ISSUE TITLE
[EXJSONUtils] Method in category conflicts with same method from another category

### ISSUE THREAD

**Issue by reichhartd** [expo/expo#21674]
### Minimal reproducible example

https://github.com/reichhartd/expo-exjsonutils-bug-example

### Summary

After I installed `expo-updates`, another SDK stopped working on iOS. During the build, a new warning was added:
```
⚠️  ld: method '-stringForKey:' in category from /Users/reichhartd/Library/Developer/Xcode/DerivedData/tokenstreetStage-djoqxbopudjjtegrjvworticckar/Build/Products/Debug-iphonesimulator/EXJSONUtils/EXJSONUtils.framework/EXJSONUtils(NSDictionary+EXJSONUtils.o) conflicts with same method from another category
```
According to the [Podfile.lock](https://github.com/expo/expo/blob/main/apps/native-tests/ios/Podfile.lock) `EXJSONUtils` is only used by `EXUpdates`. Since @wschurman is in the process of migrating `EXUpdates` and `EXManifests` to Swift, would it fix the bug to also migrate `EXJSONUtils` to Swift?

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.14.2 - /opt/homebrew/opt/node@18/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.5.0 - /opt/homebrew/opt/node@18/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
      Android SDK:
        API Levels: 31, 32, 33
        Build Tools: 30.0.3, 31.0.0, 32.0.0, 33.0.0
        System Images: android-33 | Android TV ARM 64 v8a, android-33 | Android TV Intel x86 Atom, android-33 | Google TV ARM 64 v8a, android-33 | Google TV Intel x86 Atom, android-33 | Google APIs ARM 64 v8a, android-33 | Google APIs Intel x86 Atom_64, android-33 | Google Play ARM 64 v8a, android-33 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: Electric Eel 2022.1.1 Patch 2 Electric Eel 2022.1.1 Patch 2
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.6 => 48.0.6 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
    npmGlobalPackages:
      eas-cli: 3.5.2
      expo-cli: 6.3.0
    Expo Workflow: bare
```

**Comment by wschurman** [comment]
@reichhartd - just curious, what library does this conflict with?

FWIW, the dependency was removed during the swift conversion.

**Comment by wschurman** [comment]
> When will the new version be published?

Most likely not until the next SDK release since it will be a major release (lots of breaking changes for people using the classes directly due to swift conversion).

**Comment by reichhartd** [comment]
@wschurman I have updated the minimal reproducible example to Expo 49 and the latest `expo-updates` version. Unfortunately, the problem still exists.

This is because `expo-updates` requires `expo-manifests`, which has `expo-json-utils` as a dependency.

Can the issue be reopened?

**Comment by wschurman** [comment]
Reopening until released.

**Comment by wschurman** [comment]
I believe this was released in expo-json-utils@0.7.1.
